File: Golang/463. Island Perimeter.go
func islandPerimeter(grid [][]int) int {
    ans := 0
    for i, slice := range grid {
        for j, island := range slice {
            if island == 1 {
                ans += addBorder(grid, i, j)
            }
        }
    }
    return ans
}

func addBorder(grid [][]int, i, j int) int {
    count := 0
    xy := map[int][]int{i-1: {j}, i: {j-1, j+1}, i+1: {j}}
    for key, values := range xy {
        for _, value := range values {
            if key < 0 || key == len(grid) || value < 0 || value == len(grid[0]) || grid[key][value] == 0 {
                count++
            }
        }
    }
    return count
}

File: Python3/463. Island Perimeter.py
class Solution:
    def islandPerimeter(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        return sum([self.addBorder(grid, i, j) for i in range(len(grid)) for j in range(len(grid[0])) if grid[i][j]])
    
    def addBorder(self, grid, i, j):
        return ((i == 0 or grid[i-1][j] == 0) + 
                (i == len(grid)-1 or grid[i+1][j] == 0) + 
                (j == 0 or grid[i][j-1] == 0) + 
                (j == len(grid[0])-1 or grid[i][j+1] == 0))
